code/game/objects/items/devices/radio/intercom.dm

```diff
@@ -82,46 +77,53 @@
 		var/turf/position = get_turf(src)
 		if(isnull(position) || !(position.z in level))
 			return FALSE
-	if(!src.listening)
+	if(!listening)
 		return FALSE
 	if(freq == FREQ_SYNDICATE)
-		if(!(src.syndie))
+		if(!(syndie))
 			return FALSE//Prevents broadcast of messages over devices lacking the encryption
 
 	return TRUE
 
 
 /obj/item/radio/intercom/Hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq, list/spans, message_mode)
-	if (message_mode == MODE_INTERCOM)
+	if(message_mode == MODE_INTERCOM)
 		return  // Avoid hearing the same thing twice
-	if(!anyai && !(speaker in ai))
-		return
-	..()
+	return ..()
 
-/obj/item/radio/intercom/process()
-	if(((world.timeofday - last_tick) > 30) || ((world.timeofday - last_tick) < 0))
-		last_tick = world.timeofday
+/obj/item/radio/intercom/emp_act(severity)
+	. = ..() // Parent call here will set `on` to FALSE.
+	update_icon()
 
-		var/area/A = get_area(src)
-		if(!A || emped)
-			on = FALSE
-		else
-			on = A.powered(AREA_USAGE_EQUIP) // set "on" to the power status
+/obj/item/radio/intercom/end_emp_effect(curremp)
+	. = ..()
+	AreaPowerCheck() // Make sure the area/local APC is powered first before we actually turn back on.
 
-		if(!on)
-			icon_state = "intercom-p"
-		else
-			icon_state = initial(icon_state)
+/obj/item/radio/intercom/update_icon()
+	. = ..()
+	if(on)
+		icon_state = initial(icon_state)
+	else
+		icon_state = "intercom-p"
+
+/**
+ * Proc called whenever the intercom's area loses or gains power. Responsible for setting the `on` variable and calling `update_icon()`.
+ *
+ * Normally called after the intercom's area recieves the `COMSIG_AREA_POWER_CHANGE` signal, but it can also be called directly.
+ * Arguments:
+ * * source - the area that just had a power change.
+ */
+/obj/item/radio/intercom/proc/AreaPowerCheck(datum/source)

```

Comment:
this arg isn't used anywhere.

Reply:
Yeah, but I've seen the source arg left it to denote its called via a signal, so I tend to do that by default.